### ISSUE TITLE
Tilted thin dipoles

### ISSUE THREAD

**Issue by tpersson** [SixTrack/SixTrack#1006]
In MAD-X it is possible to distinguish between a tilted element and a "missaligned" rotation (DPSI error in MAD-X). When this is converted to Sixtrack they are added together. This is fine for all elements except the dipoles since they are defining the coordinate system. This means that if you want to create a rotated dipole SixTrack will think there is a rotation error. 

Proposal:
In SixTrack there are two elements: horizontal and vertical thin dipoles. My proposal would be to remove the vertical dipole but add an additional rotation parameter that describes the wanted rotation of the multipole. In which fort, I really don't care. This would then follow the same logic as MAD-X and we would get rid of the unnecessary vertical dipole. In terms of performance there is basically no loss since we can do all the rotations at initialization. 

Any opinions? @rdemaria @vkbo

**Comment by rdemaria** [comment]
Correct, dpsi introduce closed orbit perturbations, while tilt doesn't. For the tracking, do you plan to sandwich an horizontal dipole in between two rotations?

**Comment by tpersson** [comment]
My proposal would be to rotate all the multipoles by the tilt. I just realized that it would add an additional term to the vertical plane which would add some extra calculations but could be overcome by keeping the horizontal bend as one element and a more general bend as another element. This would make vertical bend slower than today but I doubt that any machine will have enough vertical bends to make that a critical issue.   

The lazy option is to forbid any tilts from MAD-X when a dipole is present and in that way block the use of  it and if someone ever would need it we would implement it.

**Comment by rdemaria** [comment]
Would the sandwich at the level of the madx  conversion, sufficient for the general dipole?

**Comment by tpersson** [comment]
I am not sure what you mean with Sandwich. There are two options as I see it:

1. Add the tilt parameter to one of the many forts. Then tilt the element at init. Either make a new element or change the vertical dipole into a general dipole.

2. Convert the multipoles depending on the tilt in MAD-X. As long as it is a pure horizontal or vertical we can keep the current implementation in SixTrack and tilt is no longer treated as an alignment error for multipoles (which was simply wrong).The easiest is then to simply forbid diagonal dipole. It either has to be an a0 or a b0, (they can be be tilted into each other by pi/2 but can not exist for the same element). We can also allow for a diagonal bend but then we of course also have to implement it on the SixTrack side.   

Option 2 is less work in particular if we don't allow for a diagonal dipole. So that would be my recommendation for now. I am really not sure if anyone needs a diagonal dipole..

**Comment by rdemaria** [comment]
I meant to covert a diagonal dipole with a triple consisting a z-rotation element of +tilt, an h dipole, and a z-rotation of -tilt.

**Comment by tpersson** [comment]
That is also a possibility but if I understand what you mean we could have to rotate the particle twice.  Imagine we have a0 so we rotate the by pi/2 to get a h-dipole, calculate the effect, rotate back. The rest of the effect of the multipole shouldn't be calculated in that frame. However, perhaps there are some misalignment, then we have to rotate again with a different angle to account for the alignments. 
I think it would be nice to then also make check outside the particle loop if there is a tilt and if not avoid including the extra rotation. 

However, all these rotations are not needed. In principle we could take into all the alignment errors and tilt at initialization. This is a bit of work but should speed up the code (even if probably marginally). 

Maybe we can discuss next week?
